repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py

```diff
@@ -1,91 +1,109 @@
-                if repo_map.from_id in enabled_repos:
-                    rhel_repos.append(RHELTargetRepository(repoid=repo_map.to_id))
-
-        skipped_repos = enabled_repos.difference(mapped_repos)
-
-        used = {}
-        for used_repos in self.consume(UsedRepositories):
-            for used_repo in used_repos.repositories:
-                used[used_repo.repository] = used_repo.packages
-                for repo in repo_file.data:
-                    enabled_repos.add(repo.repoid)
-
-        skipped_repos = skipped_repos.intersection(set(used.keys()))
-
-        if skipped_repos:
-            pkgs = set()
-            for repo in skipped_repos:
-                pkgs.update(used[repo])
-            self.produce(SkippedRepositories(repos=list(skipped_repos), packages=list(pkgs)))
-
-        for task in self.consume(RepositoriesSetupTasks):
-            for repo in task.to_enable:
-                rhel_repos.append(RHELTargetRepository(repoid=repo))
-
-        repos_blacklisted = set()
+                        self.enabled_repos.add(repo.repoid)
+
+    def get_mapping_of_repositories(self):
+        """Get the mapping of RHEL 7 to RHEL 8 repos."""
+        for mapping_of_repositories_msg in self.consume(RepositoriesMap):
+            for repo_mapping in mapping_of_repositories_msg.repositories:
+                self.mapping_of_repositories.append(repo_mapping)
+
+    def get_target_repos(self):
+        """Get RHEL 8 repos to enable based on the mapping of the enabled RHEL 7 repos."""
+        for repo_mapping in self.mapping_of_repositories:
+            # Check if the repo mapping architecture matches the current system architecture
+            if platform.machine() != repo_mapping.arch:
+                continue
+
+            if repo_mapping.from_id in self.enabled_repos:
+                self.mapped_rhel7_repoids.add(repo_mapping.from_id)
+                self.target_repos.append(RHELTargetRepository(repoid=repo_mapping.to_id))
+
+    def process_skipped_repos(self):
+        """Send a message about any used repo that is missing in the repo mapping."""
+        self.get_used_repos()
+        not_mapped_repos = [
+            used_repo.repoid for used_repo in self.used_repos if used_repo.repoid not in self.mapped_rhel7_repoids]
+
+        if not_mapped_repos:
+            pkgs = self.get_pkgs_installed_from_not_mapped_repos(not_mapped_repos)
+            self.produce(SkippedRepositories(repos=list(not_mapped_repos), packages=list(pkgs)))
+
+    def get_used_repos(self):
+        """Get those enabled repositories from which at least one RH-signed package is currently installed."""
+        for used_repos_msg in self.consume(UsedRepositories):
+            self.used_repos += used_repos_msg.repositories
+
+    def get_pkgs_installed_from_not_mapped_repos(self, not_mapped_repos):
+        pkgs = set()
+        for repo in not_mapped_repos:
+            pkgs.update(self.used_repos[repo])
```

Comment:
I got this error:
``
    pkgs.update(self.used_repos[repo])
TypeError: list indices must be integers, not unicode
``

Reply:
Thanks! Fixed.